app/models/shelter.rb

```diff
@@ -16,6 +16,17 @@ def self.order_by_number_of_pets
       .order("pets_count DESC")
   end
 
+  def self.order_by_reverse_alphabetical
+    find_by_sql("SELECT * FROM shelters ORDER BY name DESC")
+  end
+
+  def self.shelters_with_pending_applications
+    joins(pets: { pet_applications: :application })
```

Comment:
This is a fun one for feedback! First I'm going to talk through what you're doing here and then I'll show you an easier option (but this one works great, too!)
First, you're joining the shelters table to pets, and then joining the pet_applications table to that temporary table, and then finally joining the applications table on top of that. You're doing a multi-part join here. This temporary table is HUGE and has all the data from shelters, pets, pet_applications and applications - basically every table we care about in this project. 

From there, we're filtering the data down to those rows where the application `status` column is `Pending`. We had to do this complex joining in the first step, because the shelter table alone didn't have any information about applications, so we had to do all those joins in order to arrive at applications, where this data lives. 

Because shelters have many pets, and pets can belong to many applications, this temporary joins table probably has MANY duplicate shelters. Anytime a pet from this shelter has an application associated to it, this shelter appears again. We want all the names of the shelters, but we can remove the duplicates. 

Then we're just plucking the name column to end up with an array of shelter names.

Now, a very cool AR trick here is to use associations in order to avoid having to do multi-joins like this. I call it association "mooching". We know that the `Pet` model has these asssociations: `has_many pet_applications` and `has_many :applications, through: :pets`. Since `Pet` belongs to `Shelter`, `Shelter` has permission to mooch off of these associations that its child object has. You can, therefore, add these additional associations to your Shelter model in order to leverage the associations Shelter will get from Pet:
`has_many :applications, through: :pets`

We could include `pet_applications` too but what we care about with this story is `application` status so we can just cut to the chase and go right to applications. What this says to Rails is: "Since Pet is my child object, and Pet has many applications, then I, the shelter, have many applications, too"

So now, you can do this for this query:
`self.applications.where(applications: { status: 'Pending' }).distinct.pluck(:name)`
You can also get rid of the `self` and start just with `applications`